server/src/test/java/org/elasticsearch/cluster/routing/allocation/WriteLoadConstraintMonitorTests.java

```diff
@@ -0,0 +1,552 @@
+                .values()
+                .stream()
+                .filter(stats -> nodeExceedsQueueLatencyThreshold(stats, testState.latencyThresholdMillis))
+                .count(),
+            greaterThanOrEqualTo(2L)
+        );
+
+        // Now update cluster info to add to an additional hot-spotted node
+        final AtomicBoolean thresholdIncreased = new AtomicBoolean(false);
+        var nodeUsageStatsWithExtraHotSpot = Maps.transformValues(testState.clusterInfo.getNodeUsageStatsForThreadPools(), stats -> {
+            if (thresholdIncreased.get() == false && nodeExceedsQueueLatencyThreshold(stats, testState.latencyThresholdMillis) == false) {
+                thresholdIncreased.set(true);
+                return new NodeUsageStatsForThreadPools(
+                    stats.nodeId(),
+                    Maps.transformValues(
+                        stats.threadPoolUsageStatsMap(),
+                        tpStats -> new NodeUsageStatsForThreadPools.ThreadPoolUsageStats(
+                            tpStats.totalThreadPoolThreads(),
+                            tpStats.averageThreadPoolUtilization(),
+                            testState.latencyThresholdMillis + randomLongBetween(1, 100_000)
+                        )
+                    )
+                );
+            }
+            return stats;
+        });
+
+        // We should reroute again despite the clock being the same
+        writeLoadConstraintMonitor.onNewInfo(ClusterInfo.builder().nodeUsageStatsForThreadPools(nodeUsageStatsWithExtraHotSpot).build());
+        verify(testState.rerouteService).reroute(anyString(), eq(Priority.NORMAL), any());
+    }
+
+    private static ProjectId randomAdditionalProjectId(ClusterState clusterState) {
+        return ProjectId.fromId(
+            randomValueOtherThanMany(
+                projectId -> clusterState.metadata().hasProject(ProjectId.fromId(projectId)),
+                ESTestCase::randomIdentifier
+            )
+        );
+    }
+
+    private boolean nodeExceedsQueueLatencyThreshold(NodeUsageStatsForThreadPools nodeUsageStats, long latencyThresholdMillis) {
+        return nodeUsageStats.threadPoolUsageStatsMap()
+            .get(ThreadPool.Names.WRITE)
+            .maxThreadPoolQueueLatencyMillis() > latencyThresholdMillis;
+    }
+
+    private TestState createRandomTestStateThatWillTriggerReroute() {
+        final long latencyThresholdMillis = randomLongBetween(1000, 5000);
+        final int highUtilizationThresholdPercent = randomIntBetween(70, 100);
+        final int numberOfNodes = randomIntBetween(3, 10);
+        final ClusterSettings clusterSettings = createClusterSettings(
+            WriteLoadConstraintSettings.WriteLoadDeciderStatus.ENABLED,
+            latencyThresholdMillis,
+            highUtilizationThresholdPercent
+        );
+        final LongSupplier currentTimeSupplier = ESTestCase::randomNonNegativeLong;
```

Comment:
This is very unpredictable, can go backwards in time. Could we always use an AtomicLong by default?